Test in api is unstable

## Bug Report

```
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:55.765 +08:00] [INFO] [raft.go:923] ["831b306120d34f9d is starting a new election at term 3"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:55.766 +08:00] [INFO] [raft.go:729] ["831b306120d34f9d became pre-candidate at term 3"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:55.766 +08:00] [INFO] [raft.go:824] ["831b306120d34f9d received MsgPreVoteResp from 831b306120d34f9d at term 3"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:55.766 +08:00] [INFO] [raft.go:811] ["831b306120d34f9d [logterm: 3, index: 177] sent MsgPreVote request to 976239f155150dc3 at term 3"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:55.766 +08:00] [INFO] [raft.go:811] ["831b306120d34f9d [logterm: 3, index: 177] sent MsgPreVote request to aa912e71d9b93f32 at term 3"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:56.368 +08:00] [WARN] [v3_server.go:750] ["timed out waiting for read index response (local node might have slow network)"] [timeout=11s]
[2020-08-24T04:52:10.131Z] {"level":"warn","ts":"2020-08-24T12:51:57.377+0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-026b2f2f-0a61-4ebd-8c9e-40221e536a7c/127.0.0.1:13272","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:57.377 +08:00] [ERROR] [etcdutil.go:109] ["load from etcd meet error"] [error="[PD:etcdutil:ErrLoadValue] context deadline exceeded"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:57.377 +08:00] [WARN] [etcdutil.go:112] ["kv gets too slow"] [request-key=/pd/6864408351241973623/scheduler_config/] [cost=10.000810878s] [error="context deadline exceeded"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:57.377 +08:00] [WARN] [util.go:144] ["apply request took too long"] [took=9.999357644s] [expected-duration=100ms] [prefix="read-only range "] [request="key:\"/pd/6864408351241973623/scheduler_config/\" range_end:\"/pd/6864408351241973623/scheduler_config0\" limit:1000 "] [response=] [error="context canceled"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:57.377 +08:00] [ERROR] [coordinator.go:238] ["cannot load schedulers' config"] [retry-times=9] [error="context deadline exceeded"]
[2020-08-24T04:52:10.131Z] [2020/08/24 12:51:57.377 +08:00] [FATAL] [coordinator.go:241] ["cannot load schedulers' config"] [error="context deadline exceeded"] [stack="github.com/pingcap/log.Fatal\n\t/home/jenkins/agent/workspace/pd_test/go/pkg/mod/github.com/pingcap/log@v0.0.0-20200511115504-543df19646ad/global.go:59\ngithub.com/tikv/pd/server/cluster.(*coordinator).run\n\t/home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/cluster/coordinator.go:241\ngithub.com/tikv/pd/server/cluster.(*RaftCluster).runCoordinator\n\t/home/jenkins/agent/workspace/pd_test/go/src/github.com/pingcap/pd/server/cluster/cluster.go:334"]
[2020-08-24T04:52:10.131Z] FAIL	github.com/tikv/pd/server/api	110.294s
```
https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/pd_test/runs/7161/nodes/38/steps/107/log/?start=0

--- a/pkg/errs/errno.go
+++ b/pkg/errs/errno.go
@@ -43,6 +43,7 @@ var (
 // client errors
 var (
 	ErrCloseGRPCConn   = errors.Normalize("close gRPC connection failed", errors.RFCCodeText("PD:client:ErrCloseGRPCConn"))
+	ErrCloseEtcdClient = errors.Normalize("close etcd client failed", errors.RFCCodeText("PD:client:ErrCloseEtcdClient"))
 	ErrUpdateLeader    = errors.Normalize("update leader failed", errors.RFCCodeText("PD:client:ErrUpdateLeader"))
 	ErrCreateTSOStream = errors.Normalize("create TSO stream failed", errors.RFCCodeText("PD:client:ErrCreateTSOStream"))
 	ErrGetTSO          = errors.Normalize("get TSO failed", errors.RFCCodeText("PD:client:ErrGetTSO"))

--- a/pkg/errs/errno.go
+++ b/pkg/errs/errno.go
@@ -43,7 +43,6 @@ var (
 // client errors
 var (
 	ErrCloseGRPCConn   = errors.Normalize("close gRPC connection failed", errors.RFCCodeText("PD:client:ErrCloseGRPCConn"))
-	ErrCloseEtcdClient = errors.Normalize("close etcd client failed", errors.RFCCodeText("PD:client:ErrCloseEtcdClient"))
 	ErrUpdateLeader    = errors.Normalize("update leader failed", errors.RFCCodeText("PD:client:ErrUpdateLeader"))
 	ErrCreateTSOStream = errors.Normalize("create TSO stream failed", errors.RFCCodeText("PD:client:ErrCreateTSOStream"))
 	ErrGetTSO          = errors.Normalize("get TSO failed", errors.RFCCodeText("PD:client:ErrGetTSO"))
@@ -93,4 +92,9 @@ var (
 var (
 	ErrLoadValue  = errors.Normalize("load value from etcd failed", errors.RFCCodeText("PD:etcdutil:ErrLoadValue"))
 	ErrGetCluster = errors.Normalize("get cluster from remote peer failed", errors.RFCCodeText("PD:etcdutil:ErrGetCluster"))
+)
+
+// etcd errors
+var (
+	ErrCloseEtcdClient = errors.Normalize("close etcd client failed", errors.RFCCodeText("PD:etcd:ErrCloseEtcdClient"))
 )

--- a/server/cluster/coordinator.go
+++ b/server/cluster/coordinator.go
@@ -231,6 +231,12 @@ func (c *coordinator) run() {
 	)
 	for i := 0; i < maxLoadConfigRetries; i++ {
 		scheduleNames, configs, err = c.cluster.storage.LoadAllScheduleConfig()
+		select {
+		case <-c.ctx.Done():
+			log.Info("coordinator stops running")
+			return
+		default:
+		}
 		if err == nil {
 			break
 		}

--- a/server/server.go
+++ b/server/server.go
@@ -405,11 +405,17 @@ func (s *Server) Close() {
 
 	log.Info("closing server")
 
-	s.stopServerLoop()
+	if s.httpClient != nil {
+		s.httpClient.CloseIdleConnections()
+	}
 
 	if s.client != nil {
-		s.client.Close()
+		if err := s.client.Close(); err != nil {
+			log.Error("close etcd client meet error", zap.Error(err))
+		}
 	}
+
+	s.stopServerLoop()
 
 	if s.member.Etcd() != nil {
 		s.member.Close()

--- a/server/server.go
+++ b/server/server.go
@@ -38,6 +38,7 @@ import (
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/log"
 	"github.com/pingcap/sysutil"
+	"github.com/tikv/pd/pkg/errs"
 	"github.com/tikv/pd/pkg/etcdutil"
 	"github.com/tikv/pd/pkg/grpcutil"
 	"github.com/tikv/pd/pkg/logutil"
@@ -411,7 +412,7 @@ func (s *Server) Close() {
 
 	if s.client != nil {
 		if err := s.client.Close(); err != nil {
-			log.Error("close etcd client meet error", zap.Error(err))
+			log.Error("close etcd client meet error", errs.ZapError(errs.ErrCloseEtcdClient, err))
 		}
 	}
 

--- a/server/server.go
+++ b/server/server.go
@@ -406,9 +406,7 @@ func (s *Server) Close() {
 
 	log.Info("closing server")
 
-	if s.httpClient != nil {
-		s.httpClient.CloseIdleConnections()
-	}
+	s.stopServerLoop()
 
 	if s.client != nil {
 		if err := s.client.Close(); err != nil {
@@ -416,7 +414,9 @@ func (s *Server) Close() {
 		}
 	}
 
-	s.stopServerLoop()
+	if s.httpClient != nil {
+		s.httpClient.CloseIdleConnections()
+	}
 
 	if s.member.Etcd() != nil {
 		s.member.Close()

